python/semantic_kernel/connectors/memory/qdrant/qdrant_memory_store.py

```diff
@@ -0,0 +1,416 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+"""
+QdrantMemoryStore provides functionality to add Qdrant vector database to support Semantic Kernel memory.
+The QdrantMemoryStore inherits from MemoryStoreBase for persisting/retrieving data from a Qdrant Vector Database.
+"""
+import json
+from logging import Logger
+from typing import TYPE_CHECKING, List, Optional, Tuple
+
+from numpy import ndarray
+from python.semantic_kernel.connectors.memory.qdrant.qdrant_utils import (
+    convert_from_memory_record,
+)
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.utils.null_logger import NullLogger
+
+
+if TYPE_CHECKING:
+    import qdrant_client
+    from qdrant_client.http import models
+    from qdrant_client.http.models import Distance, VectorParams, CollectionInfo
+    from qdrant_client.http.models import CollectionStatus, UpdateStatus, PointStruct
+
+
+class QdrantMemoryStore(MemoryStoreBase):
+    _qdrantclient: qdrant_client
+    _logger: Logger
+
+    def __init__(
+        self,
+        hostip: str,
+        port: Optional[int] = 6333,
+        logger: Optional[Logger] = None,
+        local: Optional[bool] = False,
+    ) -> None:
+        """Initializes a new instance of the QdrantMemoryStore class.
+
+        Arguments:
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+        try:
```

Comment:
This is a bit of a redundant check. You already have If TYPE_CHECKING: above import...

I suggest replacing the if TYPE_CHECKING block with this try: import block.

Reply:
Addressed in https://github.com/microsoft/semantic-kernel/pull/919/commits/178bc23870b345aaa73cc666148950bfc5c157c0